go/vt/vtgate/planbuilder/querytree_transformers.go

```diff
@@ -464,6 +484,13 @@ func transformRoutePlan(ctx *planningContext, n *routeTree) (*route, error) {
 }
 
 func transformJoinPlan(ctx *planningContext, n *joinTree) (logicalPlan, error) {
+	// TODO systay we should move the decision of which join to use to the greedy algorithm,
+	// and thus represented as a queryTree
+	canHashJoin, lhsInfo, rhsInfo, err := canHashJoin(ctx, n)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I see that in `canHashJoin` only `pushOutputColumns` can return an error, if that happens can we not try to use ApplyJoin?

Reply:
actually, in `canHashJoin` `removePredicate` can also return an error